metricsaggregator/src/test/java/io/dockstore/metricsaggregator/helper/ExecutionStatusAggregatorTest.java

```diff
@@ -90,8 +103,310 @@ void testGetAggregatedMetricFromWorkflowExecutions() {
+        assertEquals(1, executionStatusMetric.get().getCount().get(FAILED_RUNTIME_INVALID.toString()).getExecutionStatusCount());
+        // Submit one successful workflow execution and a list of task executions where there are two tasks that failed due to different reasons
+        taskExecutionsForOneWorkflowRun = new TaskExecutions().taskExecutions(
+                List.of(new RunExecution().executionStatus(FAILED_RUNTIME_INVALID),
+                        new RunExecution().executionStatus(FAILED_SEMANTIC_INVALID)));
+        allSubmissions = new ExecutionsRequestBody().runExecutions(List.of(submittedRunExecution)).taskExecutions(List.of(taskExecutionsForOneWorkflowRun));
+        executionStatusMetric = executionStatusAggregator.getAggregatedMetricFromAllSubmissions(allSubmissions);
+        assertTrue(executionStatusMetric.isPresent());
+        assertEquals(1, executionStatusMetric.get().getCount().get(SUCCESSFUL.toString()).getExecutionStatusCount());
+        // There should be 1 of either FAILED_RUNTIME_INVALID or FAILED_SEMANTIC_INVALID.
+        assertTrue(executionStatusMetric.get().getCount().containsKey(FAILED_RUNTIME_INVALID.toString()) || executionStatusMetric.get().getCount().containsKey(FAILED_SEMANTIC_INVALID.toString()));
+        // Submit one successful workflow execution and a list of task executions where there are 3 tasks that failed due to different reasons
+        taskExecutionsForOneWorkflowRun = new TaskExecutions().taskExecutions(
+                List.of(new RunExecution().executionStatus(FAILED_RUNTIME_INVALID),
+                        new RunExecution().executionStatus(FAILED_RUNTIME_INVALID),
+                        new RunExecution().executionStatus(FAILED_SEMANTIC_INVALID)));
+        allSubmissions = new ExecutionsRequestBody().runExecutions(List.of(submittedRunExecution)).taskExecutions(List.of(taskExecutionsForOneWorkflowRun));
+        executionStatusMetric = executionStatusAggregator.getAggregatedMetricFromAllSubmissions(allSubmissions);
+        assertTrue(executionStatusMetric.isPresent());
+        assertEquals(1, executionStatusMetric.get().getCount().get(SUCCESSFUL.toString()).getExecutionStatusCount());
+        // There should be one count of FAILED_RUNTIME_INVALID because it's the most frequent failed status in the list of task executions
+        assertEquals(1, executionStatusMetric.get().getCount().get(FAILED_RUNTIME_INVALID.toString()).getExecutionStatusCount());
+    }
+
+    @Test
+    void testGetAggregatedExecutionTime() {
+        List<RunExecution> badExecutions = new ArrayList<>();
+
+        // Add an execution that doesn't have execution time data
+        badExecutions.add(new RunExecution().executionStatus(SUCCESSFUL));
+        Optional<ExecutionStatusMetric> executionStatusMetric = EXECUTION_STATUS_AGGREGATOR.getAggregatedMetricFromAllSubmissions(new ExecutionsRequestBody().runExecutions(badExecutions));
```

Comment:
These tests were moved from ExecutionStatusAggregatorTest to here. They were slightly modified since these aggregators can't call the `getAggregatedMetricFromAllSubmissions` (have to access the metric through the execution status metric). The logic and verification is the same though.